### PR TITLE
Adds helmetcam capability to helmets

### PR DESCRIPTION
**Pull request by covertcorvid**
## About The Pull Request
Puts the squad helmetcam capability into the helmet object. Provides a way to upgrade more helmets with it, including hardsuit helmets. Converts squad helmets to being real helmets instead of hats.

## Why It's Good For The Game
These are nifty and good for reconnaissance

## Changelog
:cl:
add: Helmets can have cameras attached to them, removable by wirecutter.
code: Squad helmets are actual helmets
/:cl:

### CODE REVIEW

**Review by Vasily2013 — DISMISSED** [review]
type path changes, simple simple. Map Okay.

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 1]
```diff
@@ -660,10 +660,7 @@
 	for (var/obj/machinery/camera/C in GLOB.cameranet.cameras)
 		var/list/tempnetwork = C.network
 		//Nsv13 - AIs need to see the Rocinante.
-		//if(!(is_station_level(C.z) || is_mining_level(C.z) || ("ss13" in tempnetwork)))
-		//	continue
-		//Nsv13 end.
-		if(!SHARES_OVERMAP_ALLIED(C, src) || ("ss13" in tempnetwork))
+		if(!(is_station_level(C.z) || is_mining_level(C.z) || ("ss13" in tempnetwork)))
```
Comment (Bokkiewokkie) [inline code/modules/mob/living/silicon/ai/ai.dm:663]: Would this not allow potential Syndicate AI's to see the regular networks too?

File: nsv13/code/modules/clothing/helmet_camera.dm [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+/obj/item/clothing/head/helmet/proc/do_camera_update(oldLoc)
+	if(builtInCamera && !QDELETED(builtInCamera) && oldLoc != get_turf(loc))
+		GLOB.cameranet.updatePortableCamera(builtInCamera)
+	updating = FALSE
+
+#define SILICON_CAMERA_BUFFER 10
+/obj/item/clothing/head/helmet/proc/update_camera_location(oldLoc)
+	if(!oldLoc)
+		oldLoc = get_turf(loc)
+	oldLoc = get_turf(oldLoc)
```
[the remark this answers is not in the document]
Comment (covertcorvid) [reply nsv13/code/modules/clothing/helmet_camera.dm:64]: @Kmc2000 now that you're project lead again I can ping you to ask: why

[the remark this answers is not in the document]
Comment (TheFakeElon) [reply nsv13/code/modules/clothing/helmet_camera.dm:64]: 👏 👏 👏

**Review by Vasily2013 — APPROVED** [review]
Application tested, map changes okay. Helmet cameras do function.

**Pull request merged**